core/state_transition.go

```diff
@@ -169,8 +169,8 @@ type Message struct {
 }
 
 // TransactionToMessage converts a transaction into a Message.
-func TransactionToMessage(tx *types.Transaction, s types.Signer, baseFee *big.Int) (*Message, error) {
-	metaTxParams, err := types.DecodeAndVerifyMetaTxParams(tx)
+func TransactionToMessage(tx *types.Transaction, s types.Signer, baseFee *big.Int, isMetaTxUpgraded bool) (*Message, error) {
+	metaTxParams, err := types.DecodeAndVerifyMetaTxParams(tx, isMetaTxUpgraded)
```

Comment:
Adding parameter `isMetaTxUpgraded bool` to `TransactionToMessage` affect so many codes. However, `isMetaTxUpgraded` is just for fixing a bug in metaTx. If we need new upgrade config in `TransactionToMessage`, more parameters are required.
So I suggest to add a more general parameter here, like `chainConfig`. Thus, in the future, if we need more upgrade here, it is not necessary to add more parameters here.

Reply:
Yes we can add a configuration for easier access to the base values in case of future code changes. However, we still need to pass the corresponding real values and compare them with the configuration values. What if we change the parameter to `TransactionToMessage(xxx, config params.ChainConfig, isMetaTxUpgraded bool, inters ...interface{}) `?